src/tests/suite/daos_test_common.c

```diff
@@ -1026,3 +1026,147 @@ get_daos_prop_with_user_acl_perms(uint64_t perms)
 	D_FREE(user);
 	return prop;
 }
+
+int
+get_server_config(char *host, char *server_config_file)
+{
+	char	command[256];
+	size_t	len = 0;
+	size_t	read;
+	char	*line = NULL;
+	char	*pch;
+
+	snprintf(command, sizeof(command),
+		 "ssh %s ps ux | grep daos_server | grep start", host);
+	FILE *fp = popen(command, "r");
+
+	print_message("Command %s\n", command);
+	if (fp == NULL)
+		return -DER_INVAL;
+
+	while ((read = getline(&line, &len, fp)) != -1) {
+		print_message("line %s\n", line);
+		if (strstr(line, "--config") != NULL ||
+		    strstr(line, "-o") != NULL)
+			break;
+	}
+
+	pch = strtok(line, " ");
+	while (pch != NULL) {
+		if (strstr(pch, "yaml") != NULL &&
+		    strstr(pch, "--config") != NULL) {
+			strcpy(server_config_file, strchr(pch, '=') + 1);
+			break;
+		}
+
+		if (strstr(pch, "yaml") != NULL) {
+			strcpy(server_config_file, pch);
+			break;
+		}
+		pch = strtok(NULL, " ");
+	}
+
+	pclose(fp);
+
+	D_FREE(line);
+	return 0;
+}
+
+int verify_server_log_mask(char *host, char *server_config_file,
+			   char *log_mask){
+	char	command[256];
+	size_t	len = 0;
+	size_t	read;
+	char	*line = NULL;
+
+	snprintf(command, sizeof(command),
+		 "ssh %s cat %s", host, server_config_file);
+
+	FILE *fp = popen(command, "r");
+
+	if (fp == NULL)
+		return -DER_INVAL;
+
+	while ((read = getline(&line, &len, fp)) != -1) {
+		if (strstr(line, " log_mask") != NULL) {
+			if (strstr(line, log_mask) == NULL) {
+				print_message(
+					"Expected log_mask = %s, Found %s\n ",
+					log_mask, line);
+				return -DER_INVAL;
+			}
+		}
+	}
+
+	pclose(fp);
+	D_FREE(line);
+	return 0;
+}
+
+int get_server_log_file(char *host, char *server_config_file,
+			char *log_file)
+{
+	char	command[256];
+	size_t	len = 0;
+	size_t	read;
+	char	*line = NULL;
+
+	snprintf(command, sizeof(command),
+		 "ssh %s cat %s", host, server_config_file);
+
+	FILE *fp = popen(command, "r");
+
+	if (fp == NULL)
+		return -DER_INVAL;
+
+	while ((read = getline(&line, &len, fp)) != -1) {
+		if (strstr(line, " log_file") != NULL)
+			strcat(log_file, strrchr(line, ':') + 1);
+	}
+
+	pclose(fp);
+	D_FREE(line);
+	return 0;
+}
+
+int verify_state_in_log(char *host, char *log_file, char *state)
```

Comment:
yeah, this could be a workaround before we have client API to query detailed information of individual pool target.

Reply:
Correct, This is workaround until we get the API.

Reply:
this is not for pool target, but for device state in log right?

Reply:
Correct, This is just to open server log file and check the specific state is exist.